tests/integration/dbapi/async/test_queries_async.py

```diff
@@ -353,3 +355,46 @@ async def test_set_invalid_parameter(connection: Connection):
             await c.execute("set some_invalid_parameter = 1")
 
         assert len(c._set_parameters) == 0
+
+
+@mark.asyncio
+async def test_ss_async_execution_query(connection: Connection) -> None:
+    """Make an sql query and receive an id back."""
+    with connection.cursor() as c:
+        query_id = await c.execute(
+            "DROP TABLE IF EXISTS test_tb", [], async_execution=True
+        )
+    assert type(query_id) is str and query_id
+
+
+@mark.asyncio
+async def test_ss_async_execution_cancel(connection: Connection) -> None:
+    """Test cancel."""
+    with connection.cursor() as c:
+        query_id = await c.execute(
+            "DROP TABLE IF EXISTS test_tb", [], async_execution=True
```

Comment:
Try simulating the case where you cancel a query and validate that it was actually canceled.
I can think of something like
```
SET use_standard_sql=0
INSERT INTO tb SELECT sleepEachRow(1) FROM numbers(<seconds>)
```
Choose seconds to be not too much not to increase integration tests execution time, but enough to be able to see that it was canceled.

Reply:
And after it's canceled check that not all values were inserted (it was canceled during execution)

Reply:
I couldn't get `SET use_standard_sql=0; INSERT INTO tb SELECT sleepEachRow(1) FROM numbers(<seconds>)` to work with server-side async queries. It failed saying ss-async can't be used with multi-statement queries. And my attempts to first `SET use_standard_sql=0` in an earlier call to `execute()` also failed. Instead, I just created a long insert. At a certain point this will fail, as Firebolt gets better at inserting, but for now it's the best I can do.